File: pyg_shop_web/src/main/webapp/js/service/goodsService.js
//服务层
app.service("goodsService", function ($http) {

	//读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../goods/findAll.do');
    };
    //分页查询所有
    this.findByPage = function (pageNum,pageSize) {
        return $http.get('../goods/findByPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize);
    };
    //根据条件分页查询
    this.findByCondition = function (pageNum, pageSize, conditionObj) {
        return $http.post("../goods/findByCondition.do?pageNum=" + pageNum + "&pageSize=" + pageSize, conditionObj);
    };
    this.add = function (obj) {
        return $http.post('../goods/addCombGoods.do', obj);
    };
    //根据id查找
    this.findOneById = function (id) {
        return $http.get("../goods/findOneById.do?id=" + id);
    };
    //修改
    this.update = function (obj) {
        return $http.post("../goods/update.do", obj);
    };
    //批量删除
    this.batchDelete = function (idsArr) {
        return $http.post("../goods/batchDelete.do", idsArr);
    };



});

File: pyg_seller_interface/src/main/java/com/songxin/pyg/seller/service/GoodsService.java
package com.songxin.pyg.seller.service;
import java.util.List;
import com.songxin.pyg.pojo.TbGoods;

import com.songxin.pyg.vo.PageResultVO;
import com.songxin.pyg.vo.combvo.GoodsVO;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResultVO findByPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbGoods goods);
	
	
	/**
	 * 修改
	 */
	public void update(TbGoods goods);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbGoods findOneById(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void batchDelete(Long[] ids);

	/**
	 * 根据条件分页查询
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResultVO findByCondition(TbGoods goods, int pageNum, int pageSize);

	/**
	 * 添加货物[组合实体] 
	 * @param goods 
	 * @return void
	 * @author fishsx
	 * @date 2018/12/11 下午8:35
	 */
	public void addCombGoods(GoodsVO goods); 
	
}
